SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/TouchGrid.swift

```diff
@@ -89,13 +89,24 @@ class TouchGrid {
             }
             
             if self.curNode.childNodes.count != 0 {
+                if (self.curNode.childNodes[0] as? ButtonNode)?.dummy == true {
+                      self.curNode=self.buttonTree.rootNode!
+                }
                 //    i.e. we have not yet reached a leaf node while scanning
                 self.selectSubTree()
             }
+            
+          
+            
+            if (self.curNode as? ButtonNode)?.dummy == true {
```

Comment:
Please see above.

Reply:
Hi,in fact i tried to use dummy directly here but  there are some errors reported. will check why but let me just keep  it now